src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.Cache.cs

```diff
@@ -211,24 +217,40 @@ public JsonPropertyInfo GetProperty(ReadOnlySpan<byte> propertyName, ref ReadSta
                 }
             }
 
-            // No cached item was found. Try the main list which has all of the properties.
-
-            string stringPropertyName = JsonHelpers.Utf8GetString(propertyName);
-
+            // No cached item was found. Try the main dictionary which has all of the properties.
             Debug.Assert(PropertyCache != null);
-
-            if (!PropertyCache.TryGetValue(stringPropertyName, out info))
+            JsonPropertyInfo jsonPropertyInfo;
+            string stringPropertyName = JsonHelpers.Utf8GetString(propertyName);
+            if (PropertyCache.TryGetValue(stringPropertyName, out jsonPropertyInfo!))
```

Comment:
The `!` is a little suspect here.  I suspect the right change is to remove the `!` and then change the `JsonPropertyInfo jsonPropertyInfo;` above to instead be `JsonPropertyInfo? jsonPropertyInfo;`

Reply:
Yep. Thanks